src/Shared/FileMatcher.cs

```diff
@@ -2603,5 +2690,26 @@ private static bool DirectoryEndsWithPattern(string directoryPath, string patter
         };
 
         internal static bool IsRecursiveDirectoryMatch(string path) => path.TrimTrailingSlashes() == recursiveDirectoryMatch;
+
+        public struct GetFilesImplementationResults
+        {
+            public static readonly GetFilesImplementationResults Default = new GetFilesImplementationResults(Array.Empty<string>(), SearchAction.None, string.Empty, false);
```

Comment:
I like having a singleton for a common case, but I don't intuitively understand what `Default` means. Would `Empty` be a good name?

Reply:
Removed struct functionality but will leave comments unresolved in case struct is preferred over current implementation.